cloudsmith_cli/cli/commands/push.py

```diff
@@ -54,6 +59,79 @@ def validate_upload_file(ctx, opts, owner, repo, filepath, skip_errors):
     return md5_checksum
 
 
+def init_multipart_upload(opts, owner, repo, filepath):
+    """
+        Initialize a multipart upload.
+        Returns a tuple of (identifier, upload_url, upload_headers)
+    """
+
+    filename = os.path.basename(filepath)
+    resp = requests.post(
+        f"{FILES_API_BASE_URL}/{owner}/{repo}/",
+        headers={
+            "X-Api-Key": opts.api_key,
+            'accept': 'application/json'
+        },
+        json={
+            "method": "put_parts",
+            "filename": filename,
+            "md5_checksum": utils.calculate_file_md5(filepath)
+        }
+    )
+    resp.raise_for_status()
+
+    identifier = resp.json()["identifier"]
+    upload_url = resp.json()["upload_url"]
+    upload_headers = resp.json()["upload_headers"]
+
+    return identifier, upload_url, upload_headers
+
+
+def do_multipart_upload(opts, owner, repo, filepath, identifier, upload_url, upload_headers, progress_callback):
```

Comment:
@MrTam is this a valid comment? if yes maybe we need to consider refactoring other existing functions as it also has more than 4 args.

Reply:
It's valid, but it's because most of the other functions are using a mix of args and kwargs. I think codeclimate is really only looking at the args themselves here!

Reply:
Refactored to use **kwargs